### PR TITLE
[Yaml] parse inlined tags without values

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Comment by GuilhemN** [comment]
Tag values are always strings as they are not evaluated, I don't think this change makes sense.

File: src/Symfony/Component/Yaml/Inline.php [hunk 1]
```diff
@@ -635,16 +646,17 @@ private static function evaluateScalar($scalar, $flags, $references = array())
...
      */
-    private static function parseTag($value, &$i, $flags)
+    private static function parseTag($value, &$i, $flags, $stopOn = '')
     {
         if ('!' !== $value[$i]) {
             return;
         }
 
-        $tagLength = strcspn($value, " \t\n", $i + 1);
+        $tagLength = strcspn($value, " \t\n".$stopOn, $i + 1);
```
Comment (GuilhemN) [inline src/Symfony/Component/Yaml/Inline.php]: In fact indicators must always be forbidden in tags handle (see http://www.yaml.org/spec/1.2/spec.html#ns-tag-char).
It should be done in 3.3 as a bug fix imo.

**Comment by xabbuh** [comment]
> Tag values are always strings as they are not evaluated, I don't think this change makes sense.

I didn't find anything in the specs that forbids having a tag without a value. Did I miss anything?

**Comment by nicolas-grekas** [comment]
at least http://yaml-online-parser.appspot.com/ forbids it

**Comment by GuilhemN** [comment]
http://yaml-online-parser.appspot.com/ is bugged then :) 

> I didn't find anything in the specs that forbids having a tag without a value. Did I miss anything?

Having a tag without a value is allowed but I'm just saying the value is not `null` then but `""`.

**Comment by nicolas-grekas** [comment]
OK, it's not buggy, it's that a space is required after the tag, and then it resolves to the empty string yes.

**Comment by GuilhemN** [comment]
@nicolas-grekas the reference is http://ben-kiki.org/ypaste/data/497/index.html :)

`}` is forbidden in tag names so the parser should properly know that the tag is ended and should not need a space.

**Comment by xabbuh** [comment]
tags without values are now getting parsed as empty strings

File: src/Symfony/Component/Yaml/Inline.php [hunk 2]
```diff
@@ -635,16 +635,17 @@ private static function evaluateScalar($scalar, $flags, $references = array())
...
```
Comment (GuilhemN) [inline src/Symfony/Component/Yaml/Inline.php]: Imo this function should just be updated to:
```php
     private static function parseTag($value, &$i, $flags)
     {
         if ('!' !== $value[$i]) {
             return;
         }
 
        $tagLength = strcspn($value, " \t\n,[]{}", $i + 1);

        // ...
```

**Comment by xabbuh** [comment]
Let's wait here until we have merged #22913 so we can indeed forbid indicator characters in tags (as written in https://github.com/symfony/symfony/pull/23757#discussion_r130824148). Currently, we cannot do that because we would then break parsing the `!php/object` tag.

**Comment by GuilhemN** [comment]
https://github.com/symfony/symfony/pull/22913 and https://github.com/symfony/symfony/pull/23790 have been merged :)

**Comment by xabbuh** [comment]
rebased and ready

**Review by GuilhemN — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

**Pull request merged**